### PR TITLE
NO-ISSUE Fix GenericNack typo

### PR DESCRIPTION
**Pull request by septs**

### CODE REVIEW

**Comment by septs** [comment]
image

1. Fix a typo error, `GenerickNack` -> `GenericNack`

2. Use `binary.Read` & `binary.Write` simplify pdu header `Unmarshal ` and `Marshal`.

**Comment by septs** [comment]
@linxGnu

File: pdu/PDUHeader.go [hunk 1]
```diff
@@ -76,9 +69,5 @@ func (c *Header) SetSequenceNumber(v int32) {
 
 // Marshal to buffer.
 func (c *Header) Marshal(b *ByteBuffer) {
-	b.Grow(16)
-	b.WriteInt(c.CommandLength)
-	b.WriteInt(int32(c.CommandID))
-	b.WriteInt(int32(c.CommandStatus))
-	b.WriteInt(c.SequenceNumber)
+	_ = binary.Write(b, binary.BigEndian, c)
```
Comment (linxGnu) [inline pdu/PDUHeader.go:72]: What happen if we add another field for internal metadata. They will be marshaled to buffer? I don't think we should do that.

File: pdu/PDUHeader.go [hunk 2]
```diff
@@ -28,26 +29,18 @@ type Header struct {
...
-	if err == nil {
-		id, err = b.ReadInt()
-		if err == nil {
-			c.CommandID = data.CommandIDType(id)
-			if status, err = b.ReadInt(); err == nil {
-				c.CommandStatus = data.CommandStatusType(status)
-				c.SequenceNumber, err = b.ReadInt()
-			}
-		}
+	err = binary.Read(b, binary.BigEndian, c)
```
Comment (linxGnu) [inline pdu/PDUHeader.go:38]: Same reason with Marshaling with binary encoding

File: pdu/PDUHeader.go [hunk 3]
```diff
@@ -28,26 +29,18 @@ type Header struct {
 
 // ParseHeader parses PDU header.
 func ParseHeader(v [16]byte) (h Header) {
-	h.CommandLength = int32(binary.BigEndian.Uint32(v[:]))
-	h.CommandID = data.CommandIDType(binary.BigEndian.Uint32(v[4:]))
-	h.CommandStatus = data.CommandStatusType(binary.BigEndian.Uint32(v[8:]))
-	h.SequenceNumber = int32(binary.BigEndian.Uint32(v[12:]))
+	_ = h.Unmarshal(NewBuffer(v[:]))
```
Comment (linxGnu) [inline pdu/PDUHeader.go:32]: I really want fastest parsing here

**Review by linxGnu — CHANGES_REQUESTED** [review]
Thank you for fixing GenericNack typo. However, I think other optimization should be considered carefully.

**Review by linxGnu — APPROVED** [review]
LGTM. Thank you very much

**Pull request merged**